.github/workflows/api-test.yaml

```diff
@@ -0,0 +1,48 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run API Service Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '**' ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up AdoptOpenJDK 11
+        uses: joschi/setup-jdk@v2
```

Comment:
What's the rationale for deferring installation rather than just running it directly on the openjdk11 image?

Reply:
Nothing deep, just what's suggested in the examples and the other services we've used. e.g. https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle